compiler/rustc_ty_utils/src/layout.rs

```diff
@@ -193,7 +193,7 @@ fn layout_of_uncached<'tcx>(
         }
 
         ty::Dynamic(_, _, ty::DynStar) => {
-            let mut data = scalar_unit(Int(dl.ptr_sized_integer(), false));
+            let mut data = scalar_unit(Pointer(AddressSpace::DATA));
```

Comment:
This seems to cause issues.
```
Invalid InsertValueInst operands!
  %15 = insertvalue { ptr, ptr } undef, i64 %14, 0
in function _ZN3box13make_dyn_star17h0c15bb343f1925bdE
LLVM ERROR: Broken function found, compilation aborted!
```
Probably somewhere in the backend something still assumes an integer type here, and inserts an `i64` into this `{ ptr, ptr }`?